docs/nodes/nodes-library/nodes/Mattermost/README.md

```diff
@@ -50,13 +50,56 @@ The final workflow should look like the following image.
 
 1. First of all, you'll have to enter credentials for the Mattermost node. You can find out how to do that [here](../../../credentials/Mattermost/README.md).
-2. Select the channel you would like to post to in the *Channel ID* dropdown.
-3. Type the message you want to post in the *Message* field.
-4. Click on *Execute Node* to run the workflow.
+2. Select 'Channel' from the ***Resource*** dropdown list.
+3. Select a team from the ***Team ID*** dropdown list.
+4. Enter `Docs` in the ***Display Name*** field.
+5. Enter `docs` in the ***Name*** field.
+6. Click on ***Execute Node*** to run the node.
+
+In the screenshot below, you will notice that the node creates a new channel named `docs` in Mattermost.
+
+![Using the Mattermost node to create a channel](./Mattermost_node.png)
+
+### 3. Mattermost1 node (addUser: channel)
+
+This node will add a member to the channel that we created in the previous node.
+::: v-pre
+1. Select the credentials that you entered in the previous node.
+2. Select 'Channel' from the ***Resource*** dropdown list.
+3. Select 'Add User' from the ***Operation*** dropdown list.
+4. Click on the gears icon next to the ***Channel ID*** field and click on ***Add Expression***.
+5. Select the following in the ***Variable Selector*** section: Nodes > Mattermost > Output Data > JSON > id. You can also add the following expression: `{{$node["Mattermost"].json["id"]}}`.
+6. Select a user from the ***User ID*** dropdown list.
+7. Click on ***Execute Node*** to run the node.
+:::
+In the screenshot below, you will notice that the node adds a user to the channel that we created in the previous node.
+
+![Using the Mattermost node to add a user to a channel](./Mattermost1_node.png)
+
+### 4. Mattermost2 node (post: message)
+
+This node will post a message to the channel that we created using the Mattermost node.
+::: v-pre
+1. Select the credentials that you entered in the previous node.
+2. Click on the gears icon next to the ***Channel ID*** field and click on ***Add Expression***.
+3. Select the following in the ***Variable Selector*** section: Nodes > Mattermost > Output Data > JSON > id. You can also add the following expression: `{{$node["Mattermost"].json["id"]}}`.
+4. Enter a message in the ***Message*** field.
+5. Click on ***Execute Node*** to run the node.
+:::
+In the screenshot below, you will notice that the node posts a message to the channel that we created in the Mattermost node.
+
+![Using the Mattermost node to post a message to a channel](./Mattermost2_node.png)
+
+## FAQs
+
+### 1. How do I resolve the error for the ***Channel ID*** field?
 
+If you're not the System Administrator, you might get an error `there was a problem loading the parameter options from server: "Mattermost error response: You do not have the appropriate permissions."` next to the ***Channel ID*** field. If the system administrator has granted you the required permissions, you will be able to execute the node successfully.
```

Comment:
What are the "required" permissions here? Can you please specify?

Reply:
Also, worthwhile to mention that if the error is showing up, it doesn't necessarily mean that they won't be able to execute the node. Just means that the dropdown is not being auto populated.